ibis/backends/dask/execution/selection.py

```diff
@@ -138,25 +173,22 @@ def execute_selection_dataframe(
     sort_keys = op.sort_keys
     result = data
 
-    # Build up the individual dask structures from column expressions
     if selections:
-        # Create a unique row identifier and set it as the index. This is used
-        # in dd.concat to merge the pieces back together.
-        data = add_partitioned_sorted_column(data)
-        data_pieces = []
-        for selection in selections:
-            dask_object = compute_projection(
-                selection,
+        # if we are just performing select operations and all columns are in
+        # the table we can do a direct selection
+        if all(isinstance(s.op(), (ops.TableColumn)) for s in selections) and {
+            s.op().name for s in selections
+        }.issubset(set(result.columns)):
```

Comment:
The second condition (checking that the op names are a subset of the `result.columns` in theory should be guaranteed by the flow of execution (child relations are evaluated and passed in). Is it necessary for another reason?

Reply:
I _think_ this could be removed if I was able to work in the join selection logic https://github.com/ibis-project/ibis/blob/master/ibis/backends/dask/execution/selection.py#L82-L101 where the names have some postfixing. I can do that as a followup?

Reply:
Makes sense. No need to create an issue though, it's fine to just submit the PR right after this one.